### PR TITLE
Resolve eval false-positive for gawk

### PR DESCRIPTION
**Pull request by egibs**
Closes: #290 

This PR updates the `php_at_eval` rule to ignore the empty `eval()` string like in the `eval` rule and matches more PHP-specific patterns.

I installed `gawk` and ran the updated rule against the binary and confirmed that there were no matching strings.

### CODE REVIEW

File: rules/techniques/code_eval.yara [hunk 1]
```diff
@@ -55,7 +55,8 @@ rule php_at_eval : critical {
     hash_2017_tests = "f1a947148c092a58e354e46082b0187bce0eea38fab2a7638eb268da0752657b"
     hash_2017_mybiubiubiu_0_1_1_setup = "afd6712c7c190465c459ab1049cd559e4a2f00a5e1a4e1fe063cfefc19a330ef"
   strings:
-    $at_eval = /@eval\s{0,8}\(.{0,32}/
+    $at_eval = /@\beval\s*\(\s*(\$\w+|\.\s*"[^"]*"|\.\s*'[^']*'|\w+\(\s*\))/
```
Comment (egibs) [inline rules/techniques/code_eval.yara:58]: - `\$\w+` matches variable names
- `\.\s*"[^"]*"` matches string concatenations with double quotes
- `\.\s*'[^']*'` matches string concatenations with single quotes
- `\w+\(\s*\)` matches function names

File: rules/techniques/code_eval.yara [hunk 2]
```diff
@@ -55,7 +55,8 @@ rule php_at_eval : critical {
...
+    $not_empty = "eval()"
```
Comment (egibs) [inline rules/techniques/code_eval.yara:59]: This was one of the false positives.

File: rules/techniques/code_eval.yara (hunk 1 above)
Comment (egibs) [inline rules/techniques/code_eval.yara:58]:
```suggestion
    $at_eval = /@\beval\s*\(\s*(\$\w{0,32}|\.\s*"[^"]*"|\.\s*'[^']*'|\w+\(\s*\))/
```

**Review by tstromberg — APPROVED** [review]
This expression looks kind of crazy, but let's give it a try. In the future avoid use of * in preference to {0,256}

File: rules/techniques/code_eval.yara [hunk 3]
```diff
@@ -55,7 +55,8 @@ rule php_at_eval : critical {
     hash_2017_tests = "f1a947148c092a58e354e46082b0187bce0eea38fab2a7638eb268da0752657b"
     hash_2017_mybiubiubiu_0_1_1_setup = "afd6712c7c190465c459ab1049cd559e4a2f00a5e1a4e1fe063cfefc19a330ef"
   strings:
-    $at_eval = /@eval\s{0,8}\(.{0,32}/
+    $at_eval = /@\beval\s*\(\s*(\$\w{0,32}|\.\s*"[^"]*"|\.\s*'[^']*'|\w+\(\s*\))/
```
Comment (egibs) [inline rules/techniques/code_eval.yara:58]:
```suggestion
    $at_eval = /@\beval\s{0,32}\(\s{0,32}(\$\w{0,32}|\.\s{0,32}"[^"]{0,32}"|\.\s{0,32}'[^']{0,32}'|\w+\(\s{0,32}\))/
```

**Pull request merged**